airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -74,7 +77,13 @@ def get_datasets(
 
     total_entries = session.scalars(select(func.count(DatasetModel.id))).one()
     query = select(DatasetModel)
-    if uri_pattern:
+    if dag_ids:
+        dags_list = dag_ids.split(",")
+        query = query.filter(
+            (DatasetModel.consuming_dags.any(DagScheduleDatasetReference.dag_id.in_(dags_list)))
+            | (DatasetModel.producing_tasks.any(TaskOutletDatasetReference.dag_id.in_(dags_list)))
+        )
+    elif uri_pattern:
```

Comment:
what is the expected behavior when we mention both of them?

Reply:
dag_ids are more specific than uri_pattern, so it makes sense to me to ignore the uri_pattern if both are mentioned.  

Do you have another idea for this case?

Reply:
I think it's better practice to chain these search params. A user of the API would have no idea that the query they sent isn't actually being carried out.

Reply:
That makes sense. Thanks for the advice.